src/Core/src/Fonts/FontRegistrar.cs

```diff
@@ -0,0 +1,146 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+
+namespace Microsoft.Maui
+{
+	public class FontRegistrar : IFontRegistrar
+	{
+		readonly Dictionary<string, (string Filename, string? Alias, Assembly Assembly)> _embeddedFonts =
+			new Dictionary<string, (string Filename, string? Alias, Assembly Assembly)>();
+
+		readonly Dictionary<string, (string Filename, string? Alias)> _nativeFonts =
+			new Dictionary<string, (string Filename, string? Alias)>();
+
+		readonly Dictionary<string, (bool Success, string? Path)> _fontLookupCache =
+			new Dictionary<string, (bool Success, string? Path)>();
+
+		IEmbeddedFontLoader? _fontLoader;
+
+		public FontRegistrar(IEmbeddedFontLoader? fontLoader = null)
```

Comment:
The font registrar requires the font loader (installer) because before any fonts can actually be used, it needs to be installed. So, we can't return a font that is not yet installed.